test/crypto/sign.js

```diff
@@ -50,28 +46,39 @@ describe('sign', () => {
 
 	describe('#signMessageWithSecret', () => {
 		const signedMessage = signMessageWithSecret(notSecretMessage, defaultSecret);
-
 		it('should signTransaction the message correctly', () => {
-			(signedMessage).should.be.equal(defaultSignature);
+			(signedMessage).should.have.property('message').be.equal(notSecretMessage);
+			(signedMessage).should.have.property('signature').be.equal(defaultSignature);
+			(signedMessage).should.have.property('publicKey').be.equal(defaultPublicKey);
 		});
 	});
 
 	describe('#verifyMessageWithPublicKey', () => {
-		const signedMessage = signMessageWithSecret(notSecretMessage, defaultSecret);
-		const verifyMessage = verifyMessageWithPublicKey(signedMessage, defaultPublicKey);
+		let signature;
+		beforeEach(() => {
+			signature = {
+				message: notSecretMessage,
+				publicKey: defaultPublicKey,
+				signature: defaultSignature,
+			};
+		});
+		it('should return true on valid signature verification', () => {
+			(verifyMessageWithPublicKey(signature)).should.be.true();
+		});
 
-		it('should output the original signed message', () => {
-			(verifyMessage).should.be.equal(notSecretMessage);
+		it('should throw on invalid publicKey length', () => {
+			signature.publicKey = `${defaultPublicKey}ERROR`;
+			(verifyMessageWithPublicKey.bind(null, signature)).should.throw('Invalid publicKey, expected 32-byte publicKey');
 		});
 
-		it('should detect invalid publicKeys', () => {
-			const invalidPublicKey = `${defaultPublicKey}ERROR`;
-			(verifyMessageWithPublicKey.bind(null, signedMessage, invalidPublicKey)).should.throw('Invalid publicKey, expected 32-byte publicKey');
+		it('should throw on invalid signature length', () => {
+			signature.signature = `ERROR${defaultSignature}`;
+			(verifyMessageWithPublicKey.bind(null, signature)).should.throw('Invalid signature length, expected 64-byte signature');
 		});
 
-		it('should detect not verifiable signature', () => {
-			const invalidSignedMessage = `${signMessageWithSecret(notSecretMessage, defaultSecret)}ERROR`;
-			(verifyMessageWithPublicKey.bind(null, invalidSignedMessage, defaultPublicKey)).should.throw('Invalid signature publicKey combination, cannot verify message');
+		it('should return false on invalid signature', () => {
+			signature.signature = `W${defaultSignature.slice(1, -1)}S`;
```

Comment:
This is invalid base64. Is this a test for an *invalid* signature (not valid base64 string) or a *valid but incorrect* signature (was not generated by the relevant private key for this message)?